src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1065,10 +1107,25 @@ private void ImportRawPInvoke(MethodDesc method)
 
         private void ImportCalli(int token)
         {
+            MethodSignature methodSignature = (MethodSignature)_methodIL.GetObject(token);
+            HandleCall(null, methodSignature, ILOpcode.calli, ((ExpressionEntry)_stack.Pop()).ValueAsType(LLVM.PointerType(_universalSignature, 0), _builder));
         }
 
         private void ImportLdFtn(int token, ILOpcode opCode)
         {
+            MethodDesc method = (MethodDesc)_methodIL.GetObject(token);
+
+            if (opCode == ILOpcode.ldvirtftn && method.IsVirtual && method.OwningType.IsInterface)
+            {
+                _dependencies.Add(_compilation.NodeFactory.VirtualMethodUse(method));
+            }
+            else
+            {
+                AddMethodReference(method);
+            }
+
+            var entry = new LdTokenEntry<MethodDesc>(StackValueKind.NativeInt, "ldftn", method, GetOrCreateLLVMFunction(_compilation.NameMangler.GetMangledMethodName(method).ToString()));
```

Comment:
How is this going to work if I have:

```
.method Foo()
{
    .locals init (native int)
    ldftn Foo
    stloc.0
    ret
}
```

(I'm a bit concerned about treating ldtfn as LDTOKEN on the evaluation stack; *a lot* of places will need to become enlightened about the special kind of native int that can be in the stack slot.)

Reply:
Why would it need to be special cased? The code that consumes an LdTokenEntry treats it like any other StackEntry. When the LdTokenEntry is asked to produce an LLVMValueRef of a particular type (in this case int32*), it will do so without any external intervention.

Reply:
If there's no special ldtoken behavior attached to this, I find it odd to call the stack entry LdTokenEntry.

(I'm not actually looking at how is this used, only skimming the delta and this stood out.)

Reply:
The specialized stack entries like LdTokenEntry in CppCodeGen are left-over from unfinished refactoring done by somebody. I do not think they serve any good purpose today. You may want to get rid of them for WebAssembly importer (and CppCodeGen importer too).

Reply:
Added a new FunctionPointerEntry type that has the data we actually need for this. As @hippiehunter said, we're already tracking the right llvm value if somebody tried to use the value for something other than a delegate.